frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -342,19 +356,48 @@ export const funnelLogic = kea<funnelLogicType<TimeStepOption>>({
                 return stepsDropdown
             },
         ],
-        totalConversionRate: [
-            () => [selectors.stepsWithCount],
-            (stepsWithCount) =>
-                stepsWithCount.length > 1
-                    ? calcPercentage(stepsWithCount[stepsWithCount.length - 1].count, stepsWithCount[0].count)
-                    : 0,
-        ],
         areFiltersValid: [
             () => [selectors.filters],
             (filters) => {
                 return (filters.events?.length || 0) + (filters.actions?.length || 0) > 1
             },
         ],
+        conversionMetrics: [
+            () => [selectors.stepsWithCount, selectors.histogramStep],
+            (stepsWithCount, timeStep) => {
+                if (stepsWithCount.length <= 1) {
+                    return {
+                        average: 0,
+                        sum: 0,
+                        stepRate: 0,
+                        totalRate: 0,
+                    }
+                }
+
+                const isAllSteps = timeStep.from_step === -1
+                const fromStep = isAllSteps
+                    ? getReferenceStep(stepsWithCount, FunnelStepReference.total)
+                    : stepsWithCount[timeStep.from_step]
+                const toStep = isAllSteps ? getLastFilledStep(stepsWithCount) : stepsWithCount[timeStep.to_step]
+
+                // total sum from first step
+                // note: if last step has 0 count, it makes sense to say that total time to convert is 0
+                const sum =
```

Comment:
Hmm not sure if this is the best way to compute the overall average time to convert, perhaps it's better to get this information from the backend and calculated based on all data points?

Reply:
Hm the problem I saw with that was that it would bring a lot of computational burden into the frontend if we have thousands of events. Maybe we can precalculate these on the backend? @Twixes and @paolodamico what do you think?

Reply:
Yeah yeah exactly my point, I think this could come better from the backend, but not a question for me I guess

Reply:
Solved in #5177 (whose merge target is this PR), will require a frontend adjustment due to a breaking change in data shape though.